### DISCUSSION TITLE
"Allow add" option for M2O selects so admin can disable the + button on the select

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#13408] [Archive]
I'd like to be able to disable the + that appears on the select box for M2O. Would this be a simple flag in the config of the interface?

**Comment by azrikahar** [comment]
It is currently not possible, but there is an existing feature request (discussion) at #11903 🙂

**Reply by sxsws** [reply]
ahh, I did search before posting! Nothing in that issue about disabling the add button though so not a duplicate?.

**Reply by azrikahar** [reply]
> ahh, I did search before posting! Nothing in that issue about disabling the add button though so not a duplicate?.

No worries! Yeah although it's not directly mentioned in there, the title does reflect the "Add New" (`+` button) as they both should be in the same PR if it get's implemented 👍

**Reply by sxsws** [reply]
I would really like to start taking a look at some of these 'ideas', is there a brief overview of the construction of directus in terms of code location to allow me to zone in on certain bits?

**Reply by rijkvanzanten** [reply]
I don't have a written documentation available for the exact structure of the codebase, but just for reference, this particular feature would have to be added in https://github.com/directus/directus/tree/main/app/src/interfaces/select-dropdown-m2o.

The `index.ts` file contains all the options for the interface. We can add a new one there to configure the _allow create_ / _allow new_. The option is then passed as a prop to the Vue component. 

The O2M one already has this implemented, so that should be great for inspiration: https://github.com/directus/directus/blob/966bdfc6550d417c016e4072daee628369802dc4/app/src/interfaces/list-o2m/list-o2m.vue#L111-L112

**Reply by sxsws** [reply]
wow! thanks @rijkvanzanten!
